### PR TITLE
emscripten: 3.1.41 -> 3.1.42

### PR DESCRIPTION
**Pull request by willcohen**
###### Description of changes

Updated from 3.1.41 to 3.1.42.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by willcohen** [comment]
Result of `nixpkgs-review pr 240227` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2jack</li>
    <li>faust2jaqt</li>
    <li>faust2sc</li>
    <li>open-music-kontrollers.mephisto</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>emscripten</li>
    <li>faust (faust2)</li>
    <li>faust2csound</li>
    <li>faust2firefox</li>
    <li>faust2jackrust</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
  </ul>
</details>

File: pkgs/development/compilers/emscripten/default.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "emscripten-core";
     repo = "emscripten";
-    sha256 = "sha256-yQZrAQfNZZd65a14AFObrLrBzePBDnK0J87oTEB1Dcw=";
+    sha256 = "sha256-elp/LPd9SAuVZy42Wkgb6pCbPi2GnETTfyRJqU92S0E=";
```
Comment (drupol) [inline pkgs/development/compilers/emscripten/default.nix:35]:
```suggestion
    hash = "sha256-elp/LPd9SAuVZy42Wkgb6pCbPi2GnETTfyRJqU92S0E=";
```

**Review by drupol — CHANGES_REQUESTED** [review]
A minor change to apply and it's good to go.

File: pkgs/development/compilers/emscripten/default.nix (hunk 1 above)
Comment (willcohen) [reply pkgs/development/compilers/emscripten/default.nix]: Thanks @drupol. Fixed with latest force push.

**Review by drupol — APPROVED** [review]
Ping me when tests are done, I'll merge it ! Thanks !

**Comment by willcohen** [comment]
@drupol tests look good!

**Pull request merged**